### ISSUE TITLE
Remove grippie; use CSS "resize" property

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#340]
As per title. Grippie is annoying to style.

Community docs discussion in d.org: [Remove "grippie" from textarea when the browser supports resize in CSS3](https://www.drupal.org/node/1295336)

**Comment by BWPanda** [comment]
On my textareas I see an icon in the bottom-right corner that lets me resize both vertically and horizontally. I'm guessing this is a browser feature...? I prefer that over the custom 'grippie' (especially since resizing horizontally makes grippie look silly then), so I'm in favour of removing grippie.

**Comment by opi** [comment]
Full doc about resize CSS property: https://css-tricks.com/almanac/properties/r/resize/

This is not supported on IE/EDGE browser, then a fallback polyfill still needed, and can be based on core's jQueryUI https://jqueryui.com/resizable/

**Comment by klonos** [comment]
What @opi said ☝️ 

...plus it is now 2018! What @quicksketch said about IE/Edge not supporting this is still true today, but the global market share for these browsers seems to be below 10% *combined*. Actually, lower than 5% according to caniuse.com:

<img width="351" alt="screen shot 2018-07-17 at 11 18 07 am" src="https://user-images.githubusercontent.com/2423362/42791285-2d73764e-89b3-11e8-96bc-1e0b63076f29.png">

<img width="364" alt="screen shot 2018-07-17 at 11 17 57 am" src="https://user-images.githubusercontent.com/2423362/42791288-2fb581d6-89b3-11e8-8e92-d6dbd95a0125.png">

Core jQuery resizable sounds just fine if you ask me 😄

**Comment by docwilmot** [comment]
I just realised today that I'm not seeing the grippie on textareas in Seven. When did we remove that?

**Comment by klonos** [comment]
Time for IE usage update after 4 years:

![image](https://user-images.githubusercontent.com/2423362/152068628-44297285-f819-4e9c-8061-b2f8574af517.png)

![image](https://user-images.githubusercontent.com/2423362/152068662-4af20f86-c5ff-4241-851b-05d624a9945a.png)

The combined usage is below 1% globally now. It seems that the world has moved on to modern browsers, and recent, (chromium-based) Edge versions do support `resize` properly:

![image](https://user-images.githubusercontent.com/2423362/152068856-e4d963a1-f384-4e68-8bdf-a8503062d9e1.png)

![image](https://user-images.githubusercontent.com/2423362/152068827-ecd20f0a-96da-42c7-bcc5-4fac1705f1dd.png)

**Comment by quicksketch** [comment]
> I just realised today that I'm not seeing the grippie on textareas in Seven. When did we remove that?

We removed it when a CKEditor instance is on the field, since it provides it's own grippie. It's still there for normal text areas.

**Comment by docwilmot** [comment]
Could we use `JQuery resizable` as the polyfill here, since its already in core?

**Comment by quicksketch** [comment]
@docwilmot With IE support totally dropped, I don't think we need a polyfill at all at this point. I'd love to see the PR at https://github.com/backdrop/backdrop/pull/449 revived. Looking at it now, it seems to be lacking a backwards-compatible fallback for `#resizable = TRUE`, that should automatically be accepted as `#resizable = 'vertical'`. Also it could use some more robustness to ensure only vertical, horizontal, both, TRUE, and FALSE are usable values.

**Comment by indigoxela** [comment]
Yessss, finally! Let's get rid of the grippie. :tada:

I took a look at the really old PR from 2014. Honestly, I don't think, we should completely drop the library, as some contrib or custom project still may attach it. If it's gone, we'd cause trouble for them.
We could replace the attach behavior in it with a console log with a deprecation notice. That shouldn't break things.

Test coverage might be a bit tricky, for a js-only thing.

**Comment by indigoxela** [comment]
Very annoying, loads of unrelated phpcs and spellcheck nagging. :shrug: Not sure, what to do with those. And an unrelated test failure... :zany_face: 

Anyway, a [PR](https://github.com/backdrop/backdrop/pull/4564] is available for discussion.

**Comment by quicksketch** [comment]
Yay, looks great @indigoxela! I left some comments on the PR https://github.com/backdrop/backdrop/pull/4564

**Comment by klonos** [comment]
Thanks @indigoxela 🙏🏼 ...I had a look and left some comments/suggestions as well.

**Comment by indigoxela** [comment]
Many thanks @quicksketch and @klonos for having a look, I think I've addressed all requests, ready for another round of review.

**Comment by klonos** [comment]
Thanks @indigoxela 🙏🏼 ...I reviewed the code and it LGTM 👍🏼 ...only question that remains is whether the order of the js file paths matters in `views_views_plugins()`, and if not, then better to list them alphabetically.

**Comment by quicksketch** [comment]
I don't think alphabetizing the JS files is a requirement for this PR; especially as they were only reformatted to prevent a phpcs warning. I also don't think the order matters (so we could rearrange), but it's an unrelated request to this issue.

I merged in the latest `1.x` to get the PR sandbox working. Although there are no out-of-box uses of either `both` or `horizontal` resize options, I confirmed by manually changing the class to confirm each kind of resize works as expected, with a matching cursor.

The test coverage looks great. Thanks @indigoxela!

This PR looks good to me.

**Comment by quicksketch** [comment]
Thanks @indigoxela! I have merged https://github.com/backdrop/backdrop/pull/4564 into 1.x. This seems like potentially a substantial change in the way forms work, so I have only merged it into 1.x and not 1.26.x. I'm open to cherry-picking back into 1.26.x as a UX improvement though. Thanks everyone that worked on this issue, glad to see it finally completed.

https://github.com/backdrop/backdrop/commit/58f5ce93830723ff70a432c64de275c65ceb6bc6 by @indigoxela, @hellomrcat, @opi, @klonos, @docwilmot, @quicksketch, @jenlampton, and @BWPanda.
